x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/jwt/JwtRealmSettingsTests.java

```diff
@@ -0,0 +1,374 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+package org.elasticsearch.xpack.security.authc.jwt;
+
+import org.elasticsearch.common.settings.MockSecureSettings;
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.util.concurrent.ThreadContext;
+import org.elasticsearch.core.TimeValue;
+import org.elasticsearch.env.Environment;
+import org.elasticsearch.env.TestEnvironment;
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.xpack.core.security.authc.RealmConfig;
+import org.elasticsearch.xpack.core.security.authc.RealmSettings;
+import org.elasticsearch.xpack.core.security.authc.jwt.JwtRealmSettings;
+import org.elasticsearch.xpack.core.security.authc.support.DelegatedAuthorizationSettings;
+import org.elasticsearch.xpack.core.ssl.SSLConfigurationSettings;
+import org.junit.Before;
+
+import java.util.List;
+import java.util.Set;
+
+import static org.hamcrest.Matchers.equalTo;
+
+public class JwtRealmSettingsTests extends ESTestCase {
+
+    private static final String REALM_NAME = "jwt1-realm";
+    private ThreadContext threadContext;
+
+    @Before
+    public void setupEnv() {
+        Settings globalSettings = Settings.builder().put("path.home", createTempDir()).build();
+        threadContext = new ThreadContext(globalSettings);
+    }
+
+    public void testAllSettings() {
+        final RealmConfig realmConfig = getMockWorkingSettings();
+        JwtRealmSettingsTests.validateSettings(realmConfig, JwtRealmSettings.getSettings());
+    }
+
+    public void testAllowedIssuer() {
+        final Setting.AffixSetting<String> setting = JwtRealmSettings.ALLOWED_ISSUER;
+        final String settingKey = RealmSettings.getFullSettingKey(REALM_NAME, setting);
+        for (final String rejectedValue : new String[] { null, "" }) {
+            final Exception exception = expectThrows(IllegalArgumentException.class, () -> {
+                final Settings settings = Settings.builder().put(settingKey, rejectedValue).build();
+                final String actualValue = this.buildConfig(settings).getSetting(setting);
+                fail("No exception. Expected one for " + settingKey + "=" + rejectedValue + ". Got " + actualValue + ".");
```

Comment:
The fail message has a specific purpose, it provides more exact programmatic feedback than expectThrows(). expectThrows() only logs a generic message, either for no exception or a different exception. fail() logs the exact value which led to the no exception situation. Instead of needing to reproduce and debug to code, it is printed directly in the test log.

Alternative, I could log each loop value before entering expectThrows(). However, that makes the test log more noisy. fail() has the advantage of no log noise if all work, and only the one exact condition if there is a test failure.